pkg/vsphere/vm/vm.go

```diff
@@ -588,3 +588,29 @@ func (vm *VirtualMachine) SetVCHUpdateStatus(ctx context.Context, status bool) e
 
 	return err
 }
+
+// DisableDestroy attempts to disable the VirtualMachine.Destroy_Task method.
+// When the is disabled, the VC UI will disable/grey out the VM "delete" action.
+// Requires the "Global.Disable" VC privilege.
+// The VirtualMachine.Destroy_Task method can still be invoked via the API.
+func (vm *VirtualMachine) DisableDestroy(ctx context.Context) {
+	if !vm.IsVC() {
+		return
+	}
+
+	m := object.NewAuthorizationManager(vm.Vim25())
+
+	method := []object.DisabledMethodRequest{
+		{
+			Method: "Destroy_Task",
+			Reason: "Managed by VIC Engine",
```

Comment:
maybe we can have a general method here which takes as input a list of the names of the methods to be disabled. It doesn't matter at the moment though since we only want to disable OOB deletion.